testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/ForwardedTestHelperHandler.java

```diff
@@ -0,0 +1,24 @@
+package org.jboss.as.test.integration.web.handlers;
+
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+
+
+/**
+ * @author Jan Stourac
+ */
+public class ForwardedTestHelperHandler implements HttpHandler {
+
+    private final HttpHandler next;
+
+    public ForwardedTestHelperHandler(HttpHandler next) {
+        this.next = next;
+    }
+
+
+    public void handleRequest(HttpServerExchange exchange) throws Exception {
+        exchange.getResponseSender().send(exchange.getSourceAddress() + "|" + exchange.getRequestScheme() + "|"
```

Comment:
This is not really valid, after you have used the response sender in this way the exchange will be ended, so the next handler will be operating on a completed exchange, if you want to send back this information you should do it in a header.